src/Command/PluginUnloadCommand.php

```diff
@@ -42,55 +50,42 @@ public static function defaultName(): string
      */
     public function execute(Arguments $args, ConsoleIo $io): ?int
     {
-        $plugin = $args->getArgument('plugin');
-        if (!$plugin) {
-            $io->err('You must provide a plugin name in CamelCase format.');
-            $io->err('To unload an "Example" plugin, run `cake plugin unload Example`.');
-
-            return static::CODE_ERROR;
-        }
-
-        $app = APP . 'Application.php';
-        if (file_exists($app) && $this->modifyApplication($app, $plugin)) {
-            $io->out('');
-            $io->out(sprintf('%s modified', $app));
+        $plugin = (string)$args->getArgument('plugin');
 
+        $result = $this->modifyConfigFile($plugin);
+        if ($result === null) {
             return static::CODE_SUCCESS;
         }
 
+        $io->err($result);
+
         return static::CODE_ERROR;
     }
 
     /**
-     * Modify the application class.
+     * Modify the plugins config file.
      *
-     * @param string $app Path to the application to update.
-     * @param string $plugin Name of plugin.
-     * @return bool If modify passed.
+     * @param string $plugin Plugin name.
+     * @return string|null
      */
-    protected function modifyApplication(string $app, string $plugin): bool
+    protected function modifyConfigFile(string $plugin): ?string
     {
-        $plugin = preg_quote($plugin, '/');
-        $finder = "/
-            # whitespace and addPlugin call
-            \s*\\\$this\-\>addPlugin\(
-            # plugin name in quotes of any kind
-            \s*['\"]{$plugin}['\"]
-            # method arguments assuming a literal array with multiline args
-            (\s*,[\s\\n]*\[(\\n.*|.*){0,5}\][\\n\s]*)?
-            # closing paren of method
-            \);/mx";
+        // phpcs:ignore
+        $config = @include $this->configFile;
+        $config = Hash::normalize((array)$config);
+        if (!array_key_exists($plugin, $config)) {
+            return sprintf('Plugin `%s` could not be found', $plugin);
+        }
 
-        $content = file_get_contents($app);
-        $newContent = preg_replace($finder, '', $content);
+        unset($config[$plugin]);
 
-        if ($newContent === $content) {
-            return false;
-        }
+        $contents = '<?php' . "\n" . 'return ' . var_export($config, true) . ';';
```

Comment:
This will output using old-style arrays.

Reply:
That's PHP's fault :slightly_smiling_face:. We could check for existence of `Brick\VarExporter\VarExporter` (which is a bake dependency) and use that if available.

Reply:
Or we could just add `brick/varexporter` as a dev dependency.

Reply:
I was thinking about this too. We couldn't add it as a dev dependency since that wouldn't install for apps. We would have to make it a suggested dependency or just a full dependency.

I was going to check if a barebones cakephp app already installs a compatible php-parser (not sure why VarExporters needs php-parser).

Reply:
Having an optional dependency to get modern formatting sounds ok to me.

We could also build a small scale array serializer for this kind of situation too.

Should we be moving this command to bake? That would let us have the vardumper available and it would let us have a smaller 'core' and move a code generation tool into bake with other code generation tools.

Reply:
We have other dev commands in cakephp core. Would be nice to not have them required for prod environments, but I don't know if moving just this one would help.

Reply:
Loading plugins is a core feature (albeit used only during development) and not code generation IMO, so I don't think it belongs in bake. One should be able to use the plugin loading commands even if they don't use bake.